client/src/components/Meetup/Meetup.js

```diff
@@ -1,8 +1,689 @@
+                you were rejected from the meetup.
+              </Typography>
+            </Grid>
+          </Grid>
+        );
+        break;
+      default:
+        button = (
+          <Button
+            startIcon={<PersonAddIcon />}
+            onClick={handleJoinMeetup}
+            disabled={isUpdating}
+          >
+            Join Meetup
+          </Button>
+        );
+    }
+    return (
+      <Grid item xs={2} container justify="center" alignItems="flex-start">
+        {button}
+      </Grid>
+    );
+  };
+
+  const renderMeetup = (errorStatus) => {
+    if (errorStatus || !eventDetails) {
+      return renderError(errorStatus);
+    }
+
+    return (
+      <Grid item container spacing={2}>
+        <Grid item container>
+          <Grid item xs>
+            <Typography variant="h3">{eventDetails.title}</Typography>
+            {/* add share/link copy icon here? */}
+          </Grid>
+          {renderMeetupAction()}
+        </Grid>
+        <Grid
+          item
+          container
+          direction="column"
+          spacing={1}
+          className={classes.meetupDetails}
+        >
+          <Grid item>
+            <Typography variant="body2">
+              Time:{" "}
+              {eventDetails.time.toLocaleString(locale, meetupTimeOptions)}
+            </Typography>
+          </Grid>
+          {renderLocation()}
+          <Grid item>
+            <Typography variant="body2">
+              # Attendees:{" "}
+              {`${attendees ? attendees.length : 0} out of ${
+                eventDetails.maxCapacity
+              } (min. ${eventDetails.minCapacity})`}
+            </Typography>
+          </Grid>
+          <Grid item>
+            <Typography
+              component="ul"
+              className={classes.tagList}
+              variant="body2"
+            >
+              Tags: {eventDetails.tags ? renderTags(eventDetails.tags) : "none"}
+            </Typography>
+          </Grid>
+          <Grid item>
+            <Typography className={classes.description} variant="body1">
+              {eventDetails.description}
+            </Typography>
+          </Grid>
+          {renderOrganizer()}
+          <Grid item container justify="center" spacing={1}>
+            {renderAttendees()}
+            {user &&
+              user.id === eventDetails.owner.id &&
+              renderPendingAttendees(pendingAttendees)}
+          </Grid>
+        </Grid>
+      </Grid>
+    );
+  };
+
+  return (
+    <Box className={classes.root}>
+      <Dialog open={showLogin} onClose={() => setShowLogin(false)}>
+        <DialogContent>
+          <DialogContentText>Please log in to join a meetup.</DialogContentText>
+          <DialogActions>
+            <Button onClick={() => setShowLogin(false)}>OK</Button>
+            <Button component={RouterLink} to="/login" color="primary">
+              Sign in
+            </Button>
+          </DialogActions>
+        </DialogContent>
+      </Dialog>
```

Comment:
I think we can allow non-logged in users to see stuff, just don't render any meetup actions for them to take.

Reply:
Oops, I had it so that non-logged in users could view but would be prompted to sign in when clicking Join. Seems like I broke it while refactoring stuff, fixed now.